### PR TITLE
T1332

### PR DESCRIPTION
**Pull request by k88hudson**
Titlecard overlay plugin

### CODE REVIEW

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/7d20f0be7729d86/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/7d20f0be7729d86/output.txt

Total script time: 0.17 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @k88hudson received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/3ea66791d353392/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/3ea66791d353392/output.txt

Total script time: 0.82 mins

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Received

Command **cmd_preview** from @ScottDowne received. Current queue size: 0

Live output at: http://botio.butyr.org:8000/27fff6b9cadbfbc/output.txt

**Comment by ghost** [comment]
#### From: Bot.io (butyr.org)

---

# Failed

Full output at http://botio.butyr.org:8000/27fff6b9cadbfbc/output.txt

Total script time: 0.12 mins

File: templates/shared/plugins/titlecard/popcorn.titlecard.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+   * options variable
+   */
+  end: function( event, options ) {
+    options._container.classList.remove("on");
+  },
+
+  _teardown: function( options ) {
+    var target = options._target;
+    if ( target ) {
+    target.removeChild( options._container );
```
Comment (ScottDowne) [inline templates/shared/plugins/titlecard/popcorn.titlecard.js]: Whitespace!

Comment (ScottDowne) [inline templates/shared/plugins/titlecard/popcorn.titlecard.js]: Actually, it is alright to do:

target && target.removeChild();

here.

File: templates/shared/plugins/titlecard/popcorn.titlecard.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    media = context.media;
+
+    ctxContainer.id = id || "";
+    style.position = "absolute";
+    style.left = 0;
+    style.bottom = 0;
+
+    media.parentNode.appendChild( ctxContainer );
+
+  return ctxContainer;
```
Comment (ScottDowne) [inline templates/shared/plugins/titlecard/popcorn.titlecard.js]: whitespace.

File: templates/shared/plugins/titlecard/popcorn.titlecard.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+  }
+
+  Popcorn.plugin( "titlecard", {
+
+  manifest: {
+    about: {
+    name: "Popcorn titlecard Plugin",
+    version: "0.1",
+    author: "@k88hudson"
```
Comment (ScottDowne) [inline templates/shared/plugins/titlecard/popcorn.titlecard.js]: top three lines. Whitespace.

File: templates/shared/plugins/titlecard/popcorn.titlecard.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  },
+
+  _setup: function( options ) {
+
+    var target,
+      text,
+      container = options._container = document.createElement( "div" ),
+      h2 = options.title && "<h2>" + options.title + "</h2>\n" || "",
+      h5 = options.subheading &&  "<h5>" + options.subheading + "</h5>\n" || "",
+      titleString;
```
Comment (ScottDowne) [inline templates/shared/plugins/titlecard/popcorn.titlecard.js]: Top 5 lines should be aligned with target.

File: templates/shared/plugins/titlecard/popcorn.titlecard.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+  },
+
+  /**
+   * @member text
+   * The start function will be executed when the currentTime
+   * of the video  reaches the start time provided by the
+   * options variable
+   */
+  start: function( event, options ) {
+    options._container.classList.add("on");
```
Comment (ScottDowne) [inline templates/shared/plugins/titlecard/popcorn.titlecard.js]: Whitespace around "on"

File: templates/shared/plugins/titlecard/popcorn.titlecard.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  _setup: function( options ) {
+
+    var target,
+      text,
+      container = options._container = document.createElement( "div" ),
+      h2 = options.title && "<h2>" + options.title + "</h2>\n" || "",
+      h5 = options.subheading &&  "<h5>" + options.subheading + "</h5>\n" || "",
+      titleString;
+
+    container.classList.add("title-card");
```
Comment (ScottDowne) [inline templates/shared/plugins/titlecard/popcorn.titlecard.js]: Whitespace around "title-card"

File: templates/shared/plugins/titlecard/popcorn.titlecard.js [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+  },
+
+  /**
+   * @member text
+   * The end function will be executed when the currentTime
+   * of the video  reaches the end time provided by the
+   * options variable
+   */
+  end: function( event, options ) {
+    options._container.classList.remove("on");
```
Comment (ScottDowne) [inline templates/shared/plugins/titlecard/popcorn.titlecard.js]: Same as above.

**Comment by jbuck** [comment]
Resolved by pull request #416 (Toronto, represent.)

**Pull request closed without merging**